CMakeLists.txt

```diff
@@ -5,34 +5,40 @@ project(hersh)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-set(INCLUDE_DIR include)
-set(TEST_DIR tests)
+set(SOURCE_DIR src)
+set(EXTERNAL_DIR ${PROJECT_SOURCE_DIR}/external)
```

Comment:
I don't like that we're writing to the sources here but I will fix it when I work on `ExternalProject`.

Reply:
Why not? It follows the PFL recommendation, doesn't it?

Reply:
Can you post a link to said recommendation?

Reply:
In any case, even if that's their recommendation - I prefer not to dirty the repo with things we download when we build the project. It's nicer if we keep that all in `build`.

Reply:
[$2.6](https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs#tld.external) "The external/ directory is reserved for embedding of external projects."
I guess the reasoning is that it makes it consistent that `build` does not include source code, only build results.
But I see your point, and I don't really care.